### ISSUE TITLE
Order of user attribute groups can't be controlled easily any more

### ISSUE THREAD

**Issue by jbman** [keycloak/keycloak#33338]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

user-profile

### Describe the bug

In Keycloak 24.0.7 the order groups is now selected by display name (unresolved value).
In previous versions it was controlled by user attribute order (with groups showing up twice, if attributes with same group were not put one after the other)

The order of groups is hard to control, as the display name or display name placeholder needs to be changed for all groups to achieve an order which was displayed with a previous version.

### Version

24.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

Order of groups is controllable to achieve the same order as with previous versions

### Actual behavior

Order of groups is selected by group display name. Unresolved value, e.g. ${personalData).

### How to Reproduce?

Define at least two groups like this
```
  "groups": [
    {
      "name": "personal",
      "displayHeader": "${personalData}",
      "displayDescription": "",
      "annotations": {}
    },
    {
      "name": "additional",
      "displayHeader": "${additionalData}",
      "displayDescription": "",
      "annotations": {}
    }
  ],
  ```
  
Profile screen example: 
![image](https://github.com/user-attachments/assets/9dd1c390-33c0-45cc-b8bb-9920cc5e55ac)

The order of the groups can only be changed by renaming the group display names.

**Comment by danielFesenmeyer** [comment]
I think it would be consistent to have a configurable "guiOrder" for attribute groups, in the same way it is already supported for attributes.
This way, all attributes belonging to a group could actually be grouped together.
And, by defining the "guiOrder" for the attribute groups, it would be possible to achieve the same order as before.

**Comment by jbman** [comment]
Yes. just preserving the order in the "groups" JSON array internally and providing same order to the UI would be good.
A re-ordering functionality of attribute groups in the admin console could follow.

**Comment by pedroigor** [comment]
@danielFesenmeyer Makes sense for me too. We need guiOrder in groups.

@jbman If you take into account the order in the JSON array we'll need to end up introducing the guiOrder anyways. We don't want people editing JSON configuration manually.

**Comment by danielFesenmeyer** [comment]
Seems this issue addresses exactly the same topic: https://github.com/keycloak/keycloak/issues/27994

**Comment by danielFesenmeyer** [comment]
@jbman I like this idea.

One question is still open: How to handle attributes without group? I see these options:

1. Grouped as a "virtual group" at the top, as implemented now. But then "actual group" attributes with higher order might appear below  "virtual group" attributes with lower order, **even when the top "virtual group" attribute is lower** than the top "actual group" attribute.
2. Grouped as a "virtual group" at the position of the top "virtual group". This way, "actual group" attributes with higher order might appear below  "virtual group" attributes with lower order, but **only when the top "virtual group" attribute is higher** than the top "actual group" attribute.
3.  Handle each attribute without group as its own "virtual group": Put these attributes in between the actual groups, depending on their order. Furthermore, it would be good to visualize that in the Login UI. Currently they would just look like "lower order" attributes of the previous "actual group". (I guess, the current implementation tried to avoid this by putting one "virtual group" at the top.)

**Comment by jbman** [comment]
Attributes without group could be handled as if they all were in one group "Groupless". This means this groups could appear anywhere between other groups based on the top attribute which is in "Groupless".

**Comment by sschu** [comment]
We hope to come up with a fix for this once we have more time, so reopening it for now.

**Comment by sguilhen** [comment]
@sschu I've removed the auto-expire label for now as you might be able to work on it

**Comment by bst27** [comment]
I guess the most simple approach would be to simply ignore the groups and only consider the sort order of the attributes. This would also be easy to understand for the user because "what you see is what you get" (sorted as configured in the admin UI).
